### PR TITLE
fix(bcl converter fetching for demux post postprocessing api)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

This PR removes the functionality of the demux post processing api to determine the bcl converter based on the directory structure. The approach would fail for instances where old flow cells were demultiplexed with bcl2fastq for which the folder structure is incomplete. Instead the bcl converter is determined by the sequencer type.

This PR also sets the default option of "@OPTION.BCL_CONVERTER" to none. This change is necessary to allow for automated setting of bcl converter options but to allow for manual overrides if required (i.e. demultiplexing a novaseq6000 flow cell using bcl2fastq).

### Changed

- removed tests and logic around `is_bcl2fastq_demux_folder_structure` 
- sets the default option of `@OPTION.BCL_CONVERTER` to none. 

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1349,6 +1349,16 @@ def fixture_flow_cell(bcl2fastq_flow_cell_dir: Path) -> FlowCellDirectoryData:
     )
 
 
+@pytest.fixture(name="novaseq_flow_cell_demultiplexed_with_bcl2fastq", scope="session")
+def fixture__novaseq_flow_cell_demux_with_bcl2fastq(
+    bcl_convert_flow_cell_dir: Path,
+) -> FlowCellDirectoryData:
+    """Create a Novaseq6000 flow cell object with flow cell that is demultiplexed using bcl2fastq."""
```
Comment (henrikstranneheim) [inline tests/conftest.py:1356]:
```suggestion
    """Create a Novaseq6000 flow cell object with flow cell that is demultiplexed using Bcl2fastq."""
```

File: tests/models/test_flowcell_class.py [hunk 2]
```diff
@@ -12,7 +12,7 @@ def test_get_run_parameters_when_non_existing(fixtures_dir: Path):
         / "apps"
         / "demultiplexing"
         / "demultiplexed-runs"
-        / "201203_A00689_0200_AHVKJCDRXX"
+        / "201203_D00483_0200_AHVKJCDRXX"
```
Comment (henrikstranneheim) [inline tests/models/test_flowcell_class.py:15]: Use comma instead of slash when building path

**Review by henrikstranneheim — APPROVED** [review]
Nice! 👍

File: cg/apps/sequencing_metrics_parser/api.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ def create_sample_lane_sequencing_metrics_for_flow_cell(
     flow_cell_directory: Path, bcl_converter: str
 ) -> List[SampleLaneSequencingMetrics]:
     """Parse the sequencing metrics data for the correct demultiplexing software into the sequencing statistics model."""
-    if bcl_converter == BclConverter.BCLCONVERT:
+    if bcl_converter == BclConverter.BCLCONVERT or bcl_converter == BclConverter.DRAGEN:
```
Comment (seallard) [inline cg/apps/sequencing_metrics_parser/api.py:17]: Removing the dragen enum field is out of scope for this PR, but getting rid of it would be nice 😅

**Review by seallard — APPROVED** [review]
Great! 👍 
I think this PR would have been easier to review if the non-functional changes were done in a separate PR (renaming fixtures etc)

File: tests/models/test_flowcell_class.py (hunk 2 above)
Comment (ChrOertlin) [reply tests/models/test_flowcell_class.py:15]: fixed

File: tests/conftest.py (hunk 1 above)
Comment (ChrOertlin) [reply tests/conftest.py:1356]: Fixed

**Comment by ChrOertlin** [comment]
# Passing a bcl converter arguments sets it right shown fail is expected

cg demultiplex finish temporary -b bcl2fastq 230622_A00621_0864_BH5F5YDSX7
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Running cg demultiplex.
Instantiating cgstats api
Set environment to stage
Finish flow cell 230622_A00621_0864_BH5F5YDSX7
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Fetching files with tags in [H5F5YDSX7,samplesheet]
Fetching files from version 146398
Fetching files with tags in [H5F5YDSX7,archived_sample_sheet]
Fetching files from version 146398
Flow cell already exists in status db: H5F5YDSX7.
Failed to store flow cell data: 1872 validation errors for ParsingModel[List[cg.apps.demultiplex.sample_sheet.models.FlowCellSampleNovaSeq6000Bcl2Fastq]]

# passing no bcl converter argument will choose the right one 

cg demultiplex finish temporary 230622_A00621_0864_BH5F5YDSX7
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Running cg demultiplex.
Instantiating cgstats api
Set environment to stage
Finish flow cell 230622_A00621_0864_BH5F5YDSX7
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Fetching files with tags in [H5F5YDSX7,samplesheet]
Fetching files from version 146398
Fetching files with tags in [H5F5YDSX7,archived_sample_sheet]
Fetching files from version 146398
Flow cell already exists in status db: H5F5YDSX7.
Added samples to flow cell: H5F5YDSX7.
Parsing BCLConvert metrics file: /home/proj/stage/demultiplexed-runs/230622_A00621_0864_BH5F5YDSX7/Unaligned/Reports/Quality_Metrics.csv
Parsing BCLConvert metrics file: /home/proj/stage/demultiplexed-runs/230622_A00621_0864_BH5F5YDSX7/Unaligned/Reports/Demultiplex_Stats.csv
Parsing BCLConvert metrics file: /home/proj/stage/demultiplexed-runs/230622_A00621_0864_BH5F5YDSX7/Unaligned/Reports/Adapter_Metrics.csv

**Comment by ChrOertlin** [comment]
# samplesheet create 

cg demultiplex samplesheet create 230622_A00621_0864_BH5F5YDSX7
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Running cg demultiplex.
Creating sample sheet for flow cell 230622_A00621_0864_BH5F5YDSX7
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Fetching files with tags in [H5F5YDSX7,samplesheet]
Fetching files from version 146398
Fetching files with tags in [H5F5YDSX7,archived_sample_sheet]
Fetching files from version 146398
Check if sample sheet exists
Sample sheet already exists in flow cell directory
Fetching bundle with name: H5F5YDSX7
Bundle H5F5YDSX7 already has a file with the same name as /home/proj/stage/flow_cells/230622_A00621_0864_BH5F5YDSX7/SampleSheet.csv

File: tests/conftest.py [hunk 4]
```diff
@@ -1349,6 +1349,16 @@ def fixture_flow_cell(bcl2fastq_flow_cell_dir: Path) -> FlowCellDirectoryData:
     )
 
 
+@pytest.fixture(name="novaseq_flow_cell_demultiplexed_with_bcl2fastq", scope="session")
+def fixture__novaseq_flow_cell_demux_with_bcl2fastq(
```
Comment (diitaz93) [inline tests/conftest.py:1353]: Double underscore after `fixture`
```suggestion
def fixture_novaseq_flow_cell_demux_with_bcl2fastq(
```

**Review by diitaz93 — APPROVED** [review]
Looks good 👍

**Pull request merged**